### PR TITLE
[FIX] util/snippets: correctly quote table/column in query

### PR DESCRIPTION
**Pull request by aj-fuentes**
One literal column was present which caused issues with uppercase named
columns. Redo the quoting with better tools from psycopg2.

### CODE REVIEW

**Comment by cawo-odoo** [comment]
LGTM by itself.

What about the other uses of quote_ident in this file? E.g. I find it counter-intuitive that https://github.com/odoo/upgrade-util/blob/13289de289bc502fe0b88668028408817e8a9700/src/util/snippets.py#L89 returns `table` unquoted and `column` quoted, while other "getters" in the utils (like `table_of_model`) do not quote at all. We should consistently quote identifiers at time of use in a query, otherwise it will get confusing and hard to use.
Just my 2¢.

**Comment by aj-fuentes** [comment]
> We should consistently quote identifiers at time of use in a query, otherwise it will get confusing and hard to use.

I agree, we'd be moving it to a better standard way in subsequent patches when needed. I don't want to do a big refactoring just for that.

**Comment by KangOl** [comment]
upgradeci retry with always only base mass_mailing test_themes theme_common web web_editor website website_blog website_event website_mass_mailing website_sale in 14.0 15.0 16.0 17.0

File: src/util/snippets.py [hunk 1]
```diff
@@ -64,19 +65,18 @@ def add_snippet_names_on_html_field(cr, table, column, snippets, regex):
...
+    sql_params = {"column": sql.Identifier(column), "table": sql.Identifier(table)}
     query = cr.mogrify(
-        f"""
-        SELECT id, array((SELECT regexp_matches({quote(column)}, %(regex)s, 'g'))), {column}
-          FROM {quote(table)}
-         WHERE {quote(column)} ~ %(regex)s
-        """,
+        sql.SQL(
+            """
+            SELECT id, array((SELECT regexp_matches({colum}, %(regex)s, 'g'))), {column}
```
[the remark this answers is not in the document]
Comment (aj-fuentes) [reply src/util/snippets.py]: done

**Comment by KangOl** [comment]
upgradeci retry

**Comment by aj-fuentes** [comment]
@KangOl good to go?

**Pull request closed without merging**